stores/bench_test.go

```diff
@@ -21,101 +21,88 @@ import (
-		b.ResetTimer()
-		for i := 0; i < b.N; i++ {
-			if err := db.Transaction(context.Background(), func(tx sql.DatabaseTx) error {
-				err := tx.RenameObjects(context.Background(), bucket, dirs[frand.Intn(i+1)%len(dirs)], dirs[frand.Intn(i+1)%len(dirs)], true)
-				if err != nil && !errors.Is(err, api.ErrObjectNotFound) {
-					return err
-				}
-				return nil
-			}); err != nil {
-				b.Fatal(err)
-			}
+		if err := db.Transaction(context.Background(), func(tx sql.DatabaseTx) error {
+			return tx.ArchiveContract(context.Background(), fcid, api.ContractArchivalReasonHostPruned)
+		}); err != nil {
+			b.Fatal(err)
 		}
-	})
+	}
 }
 
-// BenchmarkPrunableContractRoots benchmarks diffing the roots of a contract
-// with a given set of roots to determine which roots are prunable.
+// BenchmarkPruneHostSectors benchmarks the performance of pruning host sectors.
 //
-// 15.32 MB/s | M1 Max | cd32fad7 (diff ~2TiB of contract data per second)
-func BenchmarkPrunableContractRoots(b *testing.B) {
+// cpu: Apple M1 Max
+// BenchmarkPruneHostSectors-10                 100          12663735 ns/op        3312059.09 MB/s     1888 B/op         49 allocs/op
+func BenchmarkPruneHostSectors(b *testing.B) {
 	// define parameters
-	batchSize := int64(25600) // 100GiB of contract data
-	contractSize := 1 << 40   // 1TiB contract
-	sectorSize := 4 << 20     // 4MiB sector
-	numSectors := contractSize / sectorSize
+	numSectors := b.N * hostSectorPruningBatchSize
+	sectorSize := 4 << 20 // 4 MiB sector
+
+	// set the amount of bytes pruned in every iteration
+	b.SetBytes(hostSectorPruningBatchSize * int64(sectorSize))
 
 	// create database
 	db, err := newTestDB(context.Background(), b.TempDir())
 	if err != nil {
 		b.Fatal(err)
 	}
 
-	// prepare database
-	fcid := types.FileContractID{1}
-	roots, err := insertContractSectors(db.DB(), fcid, numSectors)
+	// prepare host
+	hk := types.PublicKey{1}
+	err = insertHost(db.DB(), hk)
 	if err != nil {
 		b.Fatal(err)
 	}
 
-	// prepare batch
-	frand.Shuffle(len(roots), func(i, j int) {
-		roots[i], roots[j] = roots[j], roots[i]
-	})
-	batch := roots[:batchSize]
+	// prepare contract
+	fcid := types.FileContractID{1}
+	_, err = insertContract(db.DB(), hk, fcid, numSectors)
+	if err != nil {
+		b.Fatal(err)
+	}
 
-	// start benchmark
 	b.ResetTimer()
-	b.SetBytes(batchSize * 32)
 	for i := 0; i < b.N; i++ {
-		if err := db.Transaction(context.Background(), func(tx sql.DatabaseTx) error {
-			indices, err := tx.PrunableContractRoots(context.Background(), fcid, batch)
-			if err != nil {
-				return err
-			} else if len(indices) != 0 {
-				return errors.New("expected no prunable roots")
-			}
-			return nil
+		var n int64
+		if err := db.Transaction(context.Background(), func(tx sql.DatabaseTx) (err error) {
+			n, err = tx.PruneHostSectors(context.Background(), hostSectorPruningBatchSize)
+			return
 		}); err != nil {
 			b.Fatal(err)
+		} else if n != hostSectorPruningBatchSize {
```

Comment:
how does this work? We insert a host, a contract for the host with sectors but we prune those sectors immediately? Shouldn't we need to delete the contract first or something to make the sectors prunable?